gears/src/types/tx/mod.rs

```diff
@@ -26,6 +26,35 @@ pub trait TxMessage:
     fn type_url(&self) -> &'static str;
 }
 
+/// Utility type that guarantees correctness of transaction messages set
+pub struct Messages<T: TxMessage>(pub(crate) Vec<T>);
+
+impl<T: TxMessage> Messages<T> {
+    pub fn new(messages: Vec<T>) -> anyhow::Result<Messages<T>> {
+        if messages.is_empty() {
+            Err(anyhow::anyhow!(
+                "transaction applies non empty set of messages"
```

Comment:
How about using a custom error for this?

Reply:
Added custom error.